SIP-Y.md

```diff
@@ -0,0 +1,58 @@
+---
+SIP: 'N/A'
+Title: Changing of the Guardians
+Author: John Light (@john-light)
+Status: Draft
+Track: Contract
+Created: 2022-04-06
+---
+
+# SIP-Y: Changing of the Guardians
+
+## Description  
+
+If approved, this proposal will signal approval for a changing of the Guardians:
+- The role of Bitocracy Guardian, which has the power to veto any proposal that goes through Bitocracy, will be held by a 3-of-9 multisig.
```

Comment:
One comment here. We briefly discussed this in the past. On the governor contract there is no function to exchange the guardian, but the guradian change can be done indirectly by replacing the governor contract (deploying a new governor with the new guardian and transferring the ownership of the timelock contract). This is not much effort on the solidity side. What I did not mention before is that it would mean some additional work on the frontend and backend side, because they would need to query historical information from both contracts. If we go ahead with this, we need to make sure that FE and BE are prepared.
The alternative is to leave the exchequer address as the guardian and replace the addresses controlling the current exchequer multisig (because exchequer would be moved to the new MS anyway).
I prefer the alternative solution.

Reply:
> the guradian change can be done indirectly by replacing the governor contract (deploying a new governor with the new guardian and transferring the ownership of the timelock contract)

Is Bitocracy currently in control of transferring ownership of the timelock contract?

>  What I did not mention before is that it would mean some additional work on the frontend and backend side, because they would need to query historical information from both contracts.

Isn't it the case that we already have multiple governor contracts to query, which is why the history shown in the [voting app](https://bitocracy.sovryn.app/proposals) only goes back to SIP-0006? Or is this for a different reason?

> The alternative is to leave the exchequer address as the guardian and replace the addresses controlling the current exchequer multisig (because exchequer would be moved to the new MS anyway).

This would mean that the new Bitocracy Guardian would not be built using Gnosis Safe, right? So this multisig could no longer use the Pause interface that we planned to develop as a [Safe App](https://help.gnosis-safe.io/en/articles/4022030-add-a-custom-safe-app)?

Reply:
> Is Bitocracy currently in control of transferring ownership of the timelock contract?

Yes.

> Isn't it the case that we already have multiple governor contracts to query, which is why the history shown in the voting app only goes back to SIP-0006? Or is this for a different reason?

SIP-0001 through SIP-0005 were voted on using NT-SOV governance system. Note: we should surface this in the frontend!

> This would mean that the new Bitocracy Guardian would not be built using Gnosis Safe, right? So this multisig could no longer use the Pause interface that we planned to develop as a Safe App?

We would have to add this to the existing Admin Panel rather than as a Safe App.

Decision:
- We will simply rotate + add members to Exchequer Multisig.
- Add Veto interface to Admin Panel with "Veto" button for initial signer and "Confirm veto" button for remaining signers with m-of-n specified.